smac/utils/multi_objective.py

```diff
@@ -0,0 +1,57 @@
+from typing import Union, List, Tuple, Optional
+import numpy as np
+
+
+def normalize_costs(
+    values: Union[np.ndarray, List, List[List]],
+    bounds: Optional[List[Tuple[float, float]]] = None,
+) -> np.ndarray:
+    """Normalizes the costs to be between 0 and 1 if no bounds are given.
+    Otherwise, the costs are normalized according to the bounds.
+
+    Parameters
+    ----------
+    values : Union[np.ndarray, List, List[List]]
+        Cost values which should be normalized.
+        If array/list is one-dimensional, it is expanded by one dimension.
+    bounds : Optional[List[Tuple[float, float]]], optional
+        Min and max bounds which should be applied to the values, by default None.
+        If bounds are None the min and max values from the data are used.
+
+    Returns
+    -------
+    np.ndarray
+        Normalized costs
+    """
+
+    if isinstance(values, list):
+        values = np.array(values)
+
+    if len(values.shape) == 1:
+        values = np.expand_dims(values, axis=0)
+
+    normalized_values = []
+    for col in range(values.shape[1]):
+        data = values[:, col].astype(float)
+
+        if bounds is not None and len(bounds) == values.shape[1]:
+            min_value = bounds[col][0]
+            max_value = bounds[col][1]
+        else:
+            min_value = np.min(data)
+            max_value = np.max(data)
+
+        denominator = max_value - min_value
+
+        # Prevent divide by zero
+        if denominator == 0:
```

Comment:
maybe don't exactly compare to 0 but `denominator < epsilon` to avoid numerical issues.
E.g. by using np.isclose

Reply:
I second this if the gap between `max_value` and `min_value` is exceptionally small, it's likely due to the cost being slightly unstable. If your costs are that sensitive then "wtf".
```python
if denominator < 1e-10:
  ...
```

Alternatively, check the denominator for this small difference and issue a warning but continue as it already does.